File: script/history.cpp
#include "history.h"
#include <readline/history.h>

#include <script/meuh.h>
#include <core/log.h>

namespace script {

  
  history::history()
    : api("history") {

    api
      .fun("clear", [&] {
	  clear_history();
	})
      .fun("load", [&](const std::string& filename) {
	  read_history( filename.c_str() );
	})
      .fun("save", [&](const std::string& filename) {
	  write_history( filename.c_str() );
	});
    
  }

  
  void history::add(const std::string& line) {
    add_history(line.c_str());
  }


}

File: share/script/history.lua
local api = require 'api'

local history = api.load('history')
local meuh = require 'meuh'

local old = {}

old.save = history.save
old.load = history.load

local default_filename = meuh.path()..'/.history'

function history.save(filename)
   filename = filename or default_filename
   old.save( filename );
end

function history.load(filename)
   filename = filename or default_filename
   old.load( filename );
end

return history